pipeline/executors/Jenkinsfile-tier-0.groovy

```diff
@@ -0,0 +1,116 @@
+                userRemoteConfigs: [[
+                    url: 'https://github.com/red-hat-storage/cephci.git'
+                ]]
+            ])
+
+            // prepare the node for executing test suites
+            sharedLib = load("${env.WORKSPACE}/pipeline/vars/lib.groovy")
+            sharedCommon = load("${env.WORKSPACE}/pipeline/vars/common.groovy")
+            sharedLib.prepareNode()
+        }
+    }
+
+    stage('Prepare-Stages') {
+        /* Prepare pipeline stages using RHCEPH version */
+        ciMap = sharedLib.getCIMessageMap()
+        buildAction = ciMap["artifact"]["build_action"]
+        testStages = sharedLib.fetchStages(buildAction, tierLevel, testResults)
+    }
+
+    parallel testStages
+
+    stage('Publish Results') {
+        /* Publish results through E-mail and Google Chat */
+        def emailTo = "ceph-qe@redhat.com"
+        withEnv([
+            "rhcephVersion=${ciMap["artifact"]["nvr"]}",
+            "composeId=${ciMap["artifact"]["version"]}",
+            "composeUrl=${ciMap["build"]["compose-url"]}",
+            "repository=${ciMap["artifact"]["build_action"]}"
+        ]) {
+
+            if ( ! (1 in testResults.values()) ) {
+                emailTo = "ceph-qe-list@redhat.com"
+                def releaseContent = sharedLib.ReadFromReleaseFile(majorVersion, minorVersion)
+                releaseContent[tierLevel]["composes"] = releaseContent[buildAction]["composes"]
+                releaseContent[tierLevel]["last-run"] = releaseContent[buildAction]["ceph-version"]
+                sharedLib.WriteToReleaseFile(majorVersion, minorVersion, releaseContent)
+            }
+
+            sharedCommon.sendEMail("${tierLevel}".toUpperCase(), testResults)
+            sharedCommon.sendGChatNotification("${tierLevel}".toUpperCase())
+        }
+    }
+
+    stage('Publish UMB') {
+        /* send UMB message */
+        def buildState
+
+        if ( buildAction == "latest" ) { 
+            buildState = "tier-1 ready"
+        } else {
+            buildState = buildAction
+        }
+
+        def artifactsMap = [
+            "artifact": [
+                "type": "product-build",
+                "name": "Red Hat Ceph Storage",
+                "version": ciMap["artifact"]["version"],
+                "nvr": ciMap["artifact"]["nvr"],
+                "phase": "testing",
+                "state": buildState,
+            ],
+            "contact": [
+                "name": "Downstream Ceph QE",
+                "email": "ceph-qe@redhat.com",
+            ],
+            "pipeline": [
+                "name": "rhceph-generic-tier-0",
+            ],
+            "test-run": [
+                "type": "tier0",
+                "result": currentBuild.currentResult,
+                "url": env.BUILD_URL,
+                "log": "${env.BUILD_URL}/console",
+            ]
+        ]
+
+        def msgContent = writeJSON returnText: true, json: artifactsMap
+        println "${msgContent}"
+
```

Comment:
Let's discuss with Chaithra what is needed from the library.

Reply:
Sure. 
IMO, It would be good to send it from test groovy only.